docs/topics/native/native-c-interop-stability.md

```diff
@@ -0,0 +1,180 @@
+
+When a Kotlin library used in your project imports a third-party native library as its implementation detail,
+your project also gets access to that native library.
+This happens because Kotlin/Native doesn't support the separation between `api` and `implementation` dependencies,
+so native libraries always end up being `api` dependencies.
+
+Using such a transitive native dependency is prone to more compatibility issues.
+For example, a change made by the Kotlin library developer might alter the Kotlin representation of the native library
+incompatibly, leading to compatibility issues when you update the Kotlin library.
+
+So, instead of relying on a transitive dependency, please configure interoperability with the same native
+library directly.
+Use another package name for the native library, similar to how described [here](#use-custom-package-name).
+
+## Future evolution of interoperability
+
+Using native libraries has its issues.
+Some of them are listed in this document.
+To fix the issues, some breaking changes might be necessary in the future.
+This fact itself contributes to the compatibility issues as well.
+
+## Using native types in library API
+
+If you publish a Kotlin library, be careful with using native types in the API of your library:
+such usages are expected to be broken at some point in the future to fix the compatibility and other issues.
+This would affect the users of your library then.
+
+Of course, there are cases when using native types in a library API is necessary because it is required for
+the library purpose.
+For example, if a Kotlin library basically provides extensions to the native library.
+But if your case doesn't require this, please avoid or limit using native types in the library API.
+
+This section covers usages of native types in library APIs specifically.
+The recommendations in the section don't apply to the application code.
+
+They also don't apply to library implementations. In other words, for a library
+```kotlin
+// Be extra careful! Native types are used in the library API.
+public fun createUIView(): UIView
+public fun handleThirdPartyNativeType(c: ThirdPartyNativeType)
+
+// Be careful as usual: native types are not used in the library API.
+internal fun createUIViewController(): UIViewController
+public fun getDate(): String = NSDate().toString()
+```
+
+## Publishing a library using a third-party native library
+
+If you publish a Kotlin library that uses third-party native libraries, here are a few things you might find useful.
+
+### Use custom package name
+Using custom package names for third-party native libraries might help to prevent compatibility issues.
+
+When importing a native library to Kotlin, it gets a Kotlin package name.
+If this package name is not unique, users of your library might have a clash: for example, if the same native
+library is imported with the same package name elsewhere in the user's project or other dependencies, those two usages
+will clash.
```

Comment:
Should we put examples of specific errors like:
```
Linking globals named 'kniprot_...': symbol multiply defined!
```
here to make this page better SEOed

I also remember one consequence of such a collision is that you will not be able to see your cinterop API, but only your dependency's cinterop API